### ISSUE TITLE
Comma-separated values in JSON strings

### ISSUE THREAD

**Issue by rvangundy** [mitchellh/packer#2045]
I often need to run ansible playbooks with the following command:

```ansible-playbook playbook.yml -t tag1,tag2```

I need to run this from Packer using the extra_arguments property. The comma messes up the JSON parsing. Is there a way to insert a comma in to a JSON string?

**Comment by sethvargo** [comment]
Hi @rvangundy 

Could you please share your packer template? The comma should not be causing issues, as long as it is within the string quotes.

Best,
Seth

**Comment by mitchellh** [comment]
Closing due to lack of response

**Comment by billputer** [comment]
For future reference, I encountered a similar issue with the ansible-local provisioner.

The following template:
```javascript
{
  "type": "ansible-local",
  "playbook_file": "../../common.yml",
  "inventory_file": "../../inventories/local",
  "extra_arguments": "--tags=accounts,common"
}
```

Resulted in this error:

```
virtualbox-iso: Executing Ansible: cd /tmp/packer-provisioner-ansible-local && ANSIBLE_FORCE_COLOR=1 PYTHONUNBUFFERED=1 ansible-playbook /tmp/packer-provisioner-ansible-local/common.yml --tags=accounts common -c local -i /tmp/packer-provisioner-ansible-local/local
virtualbox-iso: ERROR: the playbook: common could not be found
```

Instead of "--tags=accounts,common", the command was being given the parameter "--tags=accounts common". I figured out that this was happening because I was passing a string to extra_arguments, and not an array. Changing extra_arguments to an array of strings solved my problem.

```javascript
{
  "type": "ansible-local",
  "playbook_file": "../../common.yml",
  "inventory_file": "../../inventories/local",
  "extra_arguments": [
    "--tags=accounts,common"
  ]
}
```

In summary, if you pass in a string to extra_arguments, it will convert your commas to spaces.
